src/components/picker/__tests__/index.spec.tsx

```diff
@@ -100,122 +97,185 @@ describe('Picker', () => {
       });
     });
 
-    it('Test onDismiss', () => {
-      const driver = getDriver({
-        pickerModalProps: {
-          onDismiss
-        }
-      });
-
-      expect(driver.isOpen()).toBeFalsy();
+    it('Test onPress', () => {
+      const driver = getDriver({onPress});
       driver.open();
       expect(driver.isOpen()).toBeTruthy();
       driver.cancel();
       expect(driver.isOpen()).toBeFalsy();
-      expect(onDismiss).toHaveBeenCalledTimes(2); // TODO: this should be 1
+      expect(onPress).toHaveBeenCalled();
+    });
+
+    describe('Test pickerModalProps', () => {
+      it('Test onDismiss', () => {
+        const driver = getDriver({
+          pickerModalProps: {
+            onDismiss
+          }
+        });
+        driver.open();
+        expect(driver.isOpen()).toBeTruthy();
+        driver.cancel();
+        expect(driver.isOpen()).toBeFalsy();
+        expect(onDismiss).toHaveBeenCalledTimes(2); // TODO: this should be 1
+      });
+
+      // TODO: this test is not passing yet
+      // it('Test onShow passed via pickerModalProps', async () => {
+      //   const driver = getDriver({
+      //     pickerModalProps: {
+      //       onShow
+      //     }
+      //   });
+      //   expect(driver.isOpen()).toBeFalsy();
+      //   jest.useFakeTimers();
+      //   expect(driver.isOpen()).toBeTruthy();
+      //   expect(onShow).toHaveBeenCalled();
+      // });
+
+      describe('Test Modal TopBar', () => {
+        it('should close and select items when pressing on cancel button', () => {
+          const driver = getDriver({mode: 'MULTI', topBarProps: {cancelLabel: 'Cancel'}});
+          driver.open();
+          expect(driver.isOpen()).toBeTruthy();
+          driver.selectItem(countries[2].label);
+          driver.selectItem(countries[4].label);
+          driver.done();
```

Comment:
You're passing `cancel` button and the test is saying `cancel` and here you're pressing `done`, I'm confused